### PR TITLE
[MWPW-148000] [ANALYTICS] Add analytics to Modals

### PR DESCRIPTION
**Pull request by AdobeLinhart**
Analytics are automatically added to main, but are modals outside of main? Modals should have analytics automatically added as well.

- Updated modal creation function to add analytics upon modal creation.
- Update martech function to accept a string or index as the analytic value.

Resolves: [MWPW-148000](https://jira.corp.adobe.com/browse/MWPW-148000)

QA instructions
Check Test modal for daa-lh values and links for daa-ll values. 

**Test URLs:**
- Before: https://main--cc--adobecom.hlx.page/drafts/mepdev/fragments/2024/q4/modalanalytics/marquee
- After: https://main--cc--adobecom.hlx.page/drafts/mepdev/fragments/2024/q4/modalanalytics/marquee?milolibs=modalanalytics
- Psi: https://modalanalytics--milo--adobecom.hlx.page/?martech=off

### CODE REVIEW

File: libs/martech/attributes.js [hunk 1]
```diff
@@ -82,8 +82,9 @@ export function decorateDefaultLinkAnalytics(block, config) {
 }
 
 export async function decorateSectionAnalytics(section, idx, config) {
+  const id = Number.isNaN(idx) ? `s${idx + 1}` : idx;
```
Comment (markpadbe) [inline libs/martech/attributes.js:85]:
```suggestion
  const id = !Number.isNaN(idx) ? `s${idx + 1}` : idx;
```

Comment (AdobeLinhart) [reply libs/martech/attributes.js]: Updated.

Comment (markpadbe) [inline libs/martech/attributes.js:85]: sorry, I think I'm seeing issues with !Number.isNaN in the modal tests. maybe try this:
```suggestion
  const id = Number.isInteger(idx) ? `s${idx + 1}` : idx;
```

**Review by markpadbe — CHANGES_REQUESTED** [review]

**Review by qiyundai — APPROVED** [review]
LGTM

**Review by markpadbe — APPROVED** [review]

File: libs/blocks/modal/modal.js [hunk 2]
```diff
@@ -180,6 +181,10 @@ export async function getModal(details, custom) {
     }
   });
 
+  const config = getConfig();
+
+  decorateSectionAnalytics(dialog, `${id}-modal`, config);
+
   dialog.append(close);
```
Comment (vgoodric) [inline libs/blocks/modal/modal.js:188]:
```suggestion
  decorateSectionAnalytics(dialog, `${id}-modal`, getConfig());
  dialog.append(close);
```

**Pull request merged**